src/interpolate.js

```diff
@@ -0,0 +1,27 @@
+export const defaultInterpolation = /\{\{\s*(.+?)\s*\}\}/g
+
+export const needsInterpolation = (rule,regexp) => regexp.test(rule.toJSON(true));
+
+const interpolate = (subject = '', params = {}, regexp, resolver) => {
+  let shouldReplaceFull, found;
+
+  const replaced = subject.replace(regexp, (full, matched) => {
```

Comment:
This is a weird looking interpolation function, but it's been in my toolbelt for a while now. It allows you to preserve types when interpolating. e.g.

```
const subject = {
   myField: '{{a}}',
   myArray: '{{b}}'
}

interpolateDeep(
  subject,
  {
    a: 1,
    b: ['some','array']
  }
) 

// results in { myField: 1, myArray: ['some','array'] }
```